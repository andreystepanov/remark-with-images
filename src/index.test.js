import withImages from './'
import remark from 'remark'

// prettier-ignore
const data = `
  # Title

  Some text with inline ![image 1](https://images.com/1.jpg)

  Some other text...

  ![image 2](https://images.com/2.jpg)

  Rest of the text...
`

async function parse(data, options) {
  const vfile = await remark().use(withImages, options).process(data)

  return {
    markdown: vfile.contents,
    images: vfile.data.images,
  }
}

test('defined', () => {
  expect(multilineCode).toBeDefined()
  expect(typeof multilineCode).toBe('function')
})

test('throws error if replace function fails', async () => {
  await parse(data, {
    replace: async url => {
      return new Promise((resolve, reject) => {
        setTimeout(() => {
          if (url.includes('1.jpg')) {
            return reject(new Error('Upload error'))
          }

          resolve(url.replace('images.com', 'cloud.images.com'))
        }, 2000)
      })
    },
  }).catch(e => {
    expect(e.url).toMatchSnapshot()
  })
})

test('replaces all image urls', async () => {
  const { markdown, images } = await parse(data, {
    replace: async url => {
      return new Promise((resolve, reject) => {
        setTimeout(() => {
          resolve(url.replace('images.com', 'cloud.images.com'))
        }, 500)
      })
    },
  })

  expect(markdown).toMatchSnapshot()
  expect(images).toMatchSnapshot()
})
