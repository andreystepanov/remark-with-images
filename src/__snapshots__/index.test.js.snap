// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`replaces all image urls 1`] = `
"# Title

  Some text with inline ![image 1](https://cloud.images.com/1.jpg)

  Some other text...

  ![image 2](https://cloud.images.com/2.jpg)

  Rest of the text...
"
`;

exports[`replaces all image urls 2`] = `
Array [
  Object {
    "alt": "image 1",
    "original_url": "https://images.com/1.jpg",
    "url": "https://cloud.images.com/1.jpg",
  },
  Object {
    "alt": "image 2",
    "original_url": "https://images.com/2.jpg",
    "url": "https://cloud.images.com/2.jpg",
  },
]
`;

exports[`throws error if replace function fails 1`] = `"https://images.com/1.jpg"`;
